src/flamegpu/io/JSONStateReader.cu

```diff
@@ -214,21 +213,42 @@ class JSONStateReader_impl : public rapidjson::BaseReaderHandler<rapidjson::UTF8
         }
         return true;
     }
-    bool Null() { return true; }
-    bool Bool(bool b) { return processValue<bool>(b); }
-    bool Int(int i) { return processValue<int32_t>(i); }
-    bool Uint(unsigned u) { return processValue<uint32_t>(u); }
-    bool Int64(int64_t i) { return processValue<int64_t>(i); }
-    bool Uint64(uint64_t u) { return processValue<uint64_t>(u); }
-    bool Double(double d) { return processValue<double>(d); }
-    bool String(const char*, rapidjson::SizeType, bool) {
+    bool null() {
+        bool is_array = false;
+        if (mode.top() == VariableArray) {
+            mode.pop();
+            is_array = true;
+        }
+        if (mode.top() == Environment) {
+            fprintf(stderr, "Warning: JSON Environment property '%s' contains NULL, this has been interpreted as NaN (but may represent Inf).\n", lastKey.c_str());
```

Comment:
Ideally these warnings should only be emitted if verbosity is `> Quiet`. 

Should be fine, but will need to check when I implement it.